docs/arch/01-deployment-modes.md

```diff
@@ -0,0 +1,479 @@
+
+3. **Docker** (including Docker Desktop and Rancher Desktop) - Checks for Docker socket at:
+   - `$TOOLHIVE_DOCKER_SOCKET` (if set)
+   - `/var/run/docker.sock`
+   - `~/.docker/run/docker.sock` (Docker Desktop on macOS)
+   - `~/.rd/docker.sock` (Rancher Desktop on macOS)
+
+### Detached Process Model
+
+When running in detached mode (`thv run` without `--foreground`):
+
+```mermaid
+sequenceDiagram
+    participant User
+    participant THV as thv (parent)
+    participant THV2 as thv restart<br/>(detached child)
+    participant Container
+
+    User->>THV: thv run server-name
+    THV->>THV: Save RunConfig to state
+    THV->>THV2: Fork: thv restart --foreground
+    Note over THV2: Detached process<br/>with new session
+    THV->>User: Return (PID written)
+    THV2->>Container: Attach or proxy
+    Container->>THV2: MCP traffic
+    THV2->>THV2: Apply middleware
+    Note over THV2: Runs indefinitely
+```
+
+**Key Implementation**:
+- `pkg/workloads/manager.go` - `RunWorkloadDetached` method
+- Uses `exec.Command` with `SysProcAttr` to detach
+- Sets `TOOLHIVE_DETACHED=true` environment variable
+- Redirects stdout/stderr to log file: `~/.toolhive/logs/<workload>.log`
+
+### File Locations
+
+| Purpose | Path (Linux) | Path (macOS) |
+|---------|--------------|--------------|
+| State files (RunConfig) | `~/.local/state/toolhive/` | `~/Library/Application Support/toolhive/` |
+| Data files (logs, PIDs, secrets, statuses) | `~/.local/share/toolhive/` | `~/Library/Application Support/toolhive/` |
+| Config files | `~/.config/toolhive/` | `~/Library/Application Support/toolhive/` |
+| Cache files | `~/.cache/toolhive/` | `~/Library/Caches/toolhive/` |
+
+**Implementation**: Uses `adrg/xdg` package for XDG Base Directory compliance.
+
+## Local Mode: UI
+
+### Architecture
+
+```mermaid
+graph TB
+    User[User] -->|Web Browser| Studio[ToolHive Studio<br/>Web UI]
+    Studio -->|REST API| APIServer[thv serve<br/>API Server]
+    APIServer -->|Internal| Workloads[Workloads Manager]
+    Workloads -->|Runtime API| Runtime[Container Runtime<br/>Docker/Podman/Rancher]
+    Runtime -->|creates| Container[MCP Server Container]
+    Container -->|managed by| Proxy[Proxy Process]
+    Client[MCP Client] -->|HTTP| Proxy
+
+    style Studio fill:#ba68c8
+    style APIServer fill:#90caf9
+    style Proxy fill:#81c784
+    style Container fill:#ffb74d
+```
+
+### How It Works
+
+1. **User starts UI**: ToolHive Studio application launches
+
+2. **Studio spawns API server**: `thv serve`
+   - Starts HTTP API server on configurable port (default: 8080)
+   - Exposes RESTful endpoints for workload management
+
+3. **API Server (`pkg/api/server.go`)**:
+   - Handles HTTP requests from UI
+   - Delegates to Workloads Manager
+   - Returns JSON responses
+
+4. **Workload Operations**:
+   - Create: `POST /api/v1/workloads`
+   - List: `GET /api/v1/workloads`
+   - Stop: `POST /api/v1/workloads/{name}/stop`
+   - Delete: `DELETE /api/v1/workloads/{name}`
```

Comment:
API endpoint path is incorrect. The actual API uses `/api/v1beta/` prefix, not `/api/v1/`.

```suggestion
   - Delete: `DELETE /api/v1beta/workloads/{name}`
```

Evidence: `pkg/api/server.go:218-229`